WalletWasabi.Tests/HwiTests/DeviceConnectedTests/Katas.cs

```diff
@@ -0,0 +1,330 @@
+using NBitcoin;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using WalletWasabi.Hwi2;
+using WalletWasabi.Hwi2.Exceptions;
+using WalletWasabi.Hwi2.Models;
+using WalletWasabi.KeyManagement;
+using Xunit;
+
+namespace WalletWasabi.Tests.HwiTests.DeviceConnectedTests
+{
+	/// <summary>
+	/// Kata tests are intended to be run one by one.
+	/// A kata is a type of test that requires user interaction.
+	/// User interaction shall be defined in the beginning of the each kata.
+	/// Only write katas those require button push responses (eg. don't call setup on trezor.)
+	/// </summary>
+	public class Katas
+	{
+		#region SharedVariables
+
+		// Bottleneck: user action on device.
+		public TimeSpan ReasonableRequestTimeout { get; } = TimeSpan.FromMinutes(3);
+
+		#endregion SharedVariables
+
+		[Fact]
+		public async Task TrezorTKataAsync()
+		{
+			// --- USER INTERACTIONS ---
+			//
+			// Connect an already initialized device and unlock it.
+			// Run this test.
+			// displayaddress request: refuse
+			// displayaddress request: confirm
+			// displayaddress request: confirm
+			// signtx request: refuse
+			// signtx request: confirm
+			//
+			// --- USER INTERACTIONS ---
+
+			var network = Network.Main;
+			var client = new HwiClient(network);
+			using (var cts = new CancellationTokenSource(ReasonableRequestTimeout))
+			{
+				var enumerate = await client.EnumerateAsync(cts.Token);
+				Assert.Single(enumerate);
```

Comment:
`Single` already returns the element.
```suggestion
				HwiEnumerateEntry entry = Assert.Single(enumerate);
```

Reply:
gotcha